src/main/java/org/opensearch/commons/authuser/User.java

```diff
@@ -188,7 +189,10 @@ public static User parse(final String userString) {
         if ((strs.length > 3) && !Strings.isNullOrEmpty(strs[3])) {
             requestedTenant = unescapePipe(strs[3].trim());
         }
-        return new User(userName, backendRoles, roles, Arrays.asList(), requestedTenant);
+        if ((strs.length > 4) && !Strings.isNullOrEmpty(strs[4])) {
+            customAttNames.addAll(Arrays.stream(strs[4].split(",")).map(Utils::unescapePipe).toList());
```

Comment:
This block needs to be updated to do base64 deserialization since it would be base64 encoded from the security plugin. 

Let's also change `customAttNames` to a Map bc we should stores keys and values, not only the names of the attributes.

Any caller that needs to attribute names can get the keySet of the map

Reply:
Change it to a map will require additional changes for serializing the User to streamOutput, reading from streamInput and also serialization/deserializing to and from XContent.